src/types/jsx.ts

```diff
@@ -5,1126 +5,1125 @@ import { CSS } from "../css"
 /* tslint:disable no-namespace */
 declare global {
   interface NestedArray<T> extends Array<NestedArray<T> | T> {}
+}
+
+export namespace JSXInternal {
```

Comment:
Notes about this file:

In spite of the huge line count difference, there are only two real changes to this file:

1. The `JSX` namespace was moved out of the `declare global` statement above, renamed and exported as `JSXInternal` (useful inside `purview.ts` to prevent the linter complaining about shadowed variable names). This namespace name and a lot of the code changes in this PR are inspired by Preact's similar handling of its own JSX namespace. 
2. A huge number of apparently unused generic types removed on the various HTML Element type definitions, based on the fact that serially inherited `interface HTMLAttributes<T = HTMLElement>` made no use of its generic type T.

Reply:
If there is something I am unaware of in the usage of the generic type T in defining the HTMLAttributes interface, please let me know! I would be happy to revert the changes if there is something I overlooked here.

Note that I double-checked that I am using the package version of TypeScript in VSCode when this error surfaced. Perhaps no one has looked in this package for some time? Unfortunately I can't remember if the build process succeeded or failed based on this error.